libtenzir/builtins/connectors/curl.cpp

```diff
@@ -26,19 +26,63 @@ namespace tenzir::plugins {
 
 namespace {
 
+struct http_options {
+  bool json;
+  bool form;
+  bool chunked;
+  bool multipart;
+  std::string method;
+  std::vector<http::request_item> items;
+
+  friend auto inspect(auto& f, http_options& x) -> bool {
+    return f.object(x)
+      .pretty_name("tenzir.plugins.http_options")
+      .fields(f.field("json", x.json), f.field("form", x.form),
+              f.field("chunked", x.chunked), f.field("multipart", x.multipart),
+              f.field("method", x.method), f.field("items", x.items));
+  }
+};
+
 struct connector_args {
-  curl_options options;
-  std::chrono::milliseconds poll_timeout = 1s;
+  std::string url;
+  transfer_options transfer_opts;
+  http_options http_opts;
 
-  template <class Inspector>
-  friend auto inspect(Inspector& f, connector_args& x) -> bool {
+  friend auto inspect(auto& f, connector_args& x) -> bool {
     return f.object(x)
-      .pretty_name("connector_args")
-      .fields(f.field("options", x.options),
-              f.field("poll_timeout", x.poll_timeout));
+      .pretty_name("tenzir.plugins.connector_args")
+      .fields(f.field("url", x.url), f.field("transfer_opts", x.transfer_opts),
+              f.field("http_opts", x.http_opts));
   }
 };
 
+auto make_request(const connector_args& args) -> caf::expected<http::request> {
+  auto result = http::request{};
+  if (not parsers::uri(args.url, result.uri))
```

Comment:
I tried to use a url like `http://localhost` to test things out. This failed because the http spec mandates that `path` must not be empty, and contain a single "/" if it otherwise would be (https://stackoverflow.com/a/2581514).

Curl (the tool) adds that implicitly, and I believe we should do the same.
I'm not sure if it should be added here or in the `uri` parser, but I'd be willing to add the change to this PR.

Reply:
I realized that modifying the input in the parser is not really an option.

Reply:
Okay, resolving as "not actionable" in the scope of this PR.